packages/webapp/src/hooks/useOfflineCheck.ts

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect } from "react";
+import { toast } from "sonner";
+
+async function performNetworkCheck() {
+  const endpoint = "https://www.google.com";
+  const response = await fetch(endpoint, {
+    method: "HEAD",
+    cache: "no-cache",
+    mode: "no-cors",
+  });
+  return response.ok;
+}
+
+function useOfflineCheck(options: any = {}) {
+  const {
+
+    maxRetries = 3, // Maximum number of retries before assuming offline
+    retryDelay = 1500, // Delay between retries (ms)
+  } = options;
+
+  const [retries, setRetries] = useState(0);
+  
+
+  useEffect(() => {
+    const handleOffline = () => {
+      if (retries < maxRetries) {
+        setTimeout(() => {
+          performNetworkCheck()
+            .then(() => {
+              setRetries(0);
+              toast.dismiss();
+            }) // online
+            .catch(() => {
+              const newRetry = retries + 1;
+              setRetries(newRetry); // Retry if error occurs
+            });
+        }, retryDelay);
+      } 
+    };
```

Comment:
The retry logic still doesn't make sense.

1. `handleOffline` is called
2. `performNetworkCheck` is called
2. in case of failure (`catch`) the retry count is incremented by 1
3. ??? — nothing in the code makes `performNetworkCheck` or `handleOffline` be called again

Reply:
Do we really need retry logic in the first place? We could just assume offline after a single failure.

The risk of assuming after single failure is that it's a fluke and the `offline` event and subsequent `online` events are never sent, in which case we stay in offline mode forever.

Do we even need `performNetworkCheck` at all? Can't we only rely on `online` and `offline`?

It's possible that these are not always reliable and it's recommend to perform explicit checks, but I'd like to hear your reasoning about this + some documentation showing that indeed `online` / `offline` is not to be relied on.

Reply:
Hmmmm, it might be complicating things. The first option I opted for was using the `online` and `offline `, but seeing some documentation that suggests online/offline might not be relied on that's why I opted for doing a network check and retrying the request if it fails the first time.

In my opinion, the unreliability of the online/offline event might be rare and we can tackle it with listening for both online and offline. I implemented a version that uses the online and offline and they both work. I will push the changes for review. 

Cc: @norswap